promql/functions.go

```diff
@@ -1385,18 +1385,11 @@ func init() {
 		delete(parser.Functions, "xincrease")
 		delete(parser.Functions, "xrate")
 		fmt.Println("Successfully replaced rate & friends with xrate & friends (and removed xrate & friends function keys).")
-	case "2":
+	case "2", "y", "Y":
 		FunctionCalls["delta"] = FunctionCalls["ydelta"]
-		parser.Functions["delta"] = parser.Functions["ydelta"]
-		parser.Functions["delta"].Name = "delta"
```

Comment:
The `Functions` map has pointers to `Function` struct, so assigning the name like this would overwrite the shared struct I think. And in any case, it's not necessary because these structs are identical except for the name. (So I believe the code I copied from above was wasting its time doing the assignment and then name overwrite.)

The only thing that matters is the actual pointer to the function which is in the separate `FunctionCalls` map.

Reply:
Building on the above, I decided to add `parser.Functions` entries for `orig_increase`/`orig_rate`/`orig_delta` to give ourselves access to those in case we need them for debugging any surprises.

Reply:
> I believe the code I copied from above was wasting its time doing the assignment and then name overwrite.

Correct, you'd need to dereference the pointer first to create a shallow copy (like you're doing in the copy func below).